fv3core/stencils/dyn_core.py

```diff
@@ -387,17 +389,23 @@ def __init__(
             )
             self._zs = utils.make_storage_data(zs_3d[:, :, 0], zs_3d.shape[0:2], (0, 0))
             self.update_height_on_d_grid = updatedzd.UpdateHeightOnDGrid(
-                self.grid.grid_indexing,
-                self.grid.damping_coefficients,
-                self.grid.grid_data,
-                self.grid.grid_type,
+                grid_indexing,
+                damping_coefficients,
+                grid_data,
+                grid_type,
                 namelist.hord_tm,
                 self._dp_ref,
                 column_namelist,
                 d_sw.k_bounds(),
             )
-            self.riem_solver3 = RiemannSolver3(namelist)
-            self.riem_solver_c = RiemannSolverC(namelist)
+            self.riem_solver3 = RiemannSolver3(
+                grid_indexing,
+                spec.namelist.p_fac,
```

Comment:
A simple comment: as we expand our effort to rename and clarify the code, we might want to think about the relevance of the word `namelist` in a non-fortran code. I acknowledge that with time this has become a kind of climate simulation concept, but it's worth asking the question.
(One argument for keeping it being that it'll help not loose fortran dev)

Reply:
I agree, this will come up more as we refactor the namelist configuration.